unix/src/parse.c

```diff
@@ -118,6 +120,50 @@ static size_t curlDataCallback(void *contents, size_t size, size_t nmemb, void *
     return size * nmemb;
 }
 
+ParseClient parseInitializeWithServerURL(
+        const char *applicationId,
+        const char *clientKey,
+        const char *serverURL)
+{
+    parseSetLogLevel(PARSE_LOG_WARN);
+
+    ParseClientInternal *client = calloc(1, sizeof(ParseClientInternal));
+    if (client == NULL) {
+        parseLog(PARSE_LOG_ERROR, "%s:%s generated out of memory.\n", __FUNCTION__, __LINE__);
+        return NULL;
+    }
+    if (applicationId != NULL)
+        client->applicationId= strdup(applicationId);
+    if (clientKey != NULL)
+        client->clientKey = strdup(clientKey);
+    if (serverURL != NULL)
+        client->serverURL = strdup(serverURL);
+
+    char version[256];
+    parseOsGetVersion(version, sizeof(version));
+    client->osVersion = strdup(version);
+
+    char temp[40];
+    parseOsLoadKey(client->applicationId, PARSE_INSTALLATION_ID, temp, sizeof(temp));
+    if (temp[0] != '\0') {
+        parseSetInstallationId((ParseClient)client, temp);
+    }
+
+    parseOsLoadKey(client->applicationId, PARSE_SESSION_TOKEN, temp, sizeof(temp));
+    if (temp[0] != '\0') {
+        parseSetSessionToken((ParseClient)client, temp);
+    }
+
+    parseOsLoadKey(client->applicationId, PARSE_LAST_PUSH_TIME, temp, sizeof(temp));
+    if (temp[0] != '\0') {
+        client->lastPushTime = strdup(temp);
+    }
+
+    curl_global_init(CURL_GLOBAL_ALL);
+
+    return (ParseClient)client;
+}
+
 ParseClient parseInitialize(
         const char *applicationId,
```

Comment:
Refactor this method to call the one above with the default server URL. This way the initialization logic will not be duplicated.